### ISSUE TITLE
Release all affected packages?

### ISSUE THREAD

**Issue by csantos1113** [adobe/react-spectrum#2602]
# 🐛 Bug Report 

Multiple react-aria packages are falling behind versions so you can't really use multiple react-aria packages together

## 🤔 Expected Behavior

All affected packages to be released when an internal package dependency is updated.

## 😯 Current Behavior

Internal packages are pointing to old versions of other internal packages.

## 💁 Possible Solution

Publish new versions of all packages that are affected by a code change made to internal packages. Maybe implementing semantic-release and multi-semantic-release?

## 🔦 Context

Building my own design system using react-aria packages and connecting them all together.

## 💻 Code Sample

Example number 1:
```
import React, { useCallback, useRef } from "react";
import { useButton } from "@react-aria/button";
import type { PressEvent } from "@react-types/shared";


export default function App() {
  const buttonRef = useRef<HTMLButtonElement>(null);
  const onPress = useCallback<(e: PressEvent) => void>((e) => {
    console.log(e)
  }, []);
  const { buttonProps } = useButton(
    {
      onPress,
      type: 'button'
    },
    buttonRef
  );
  return (
    <button {...buttonProps}>Click Me</button>
  );
}
```
typescript complains about `onPress` type not being the same 
![image](https://user-images.githubusercontent.com/9648559/143313297-3623f76e-14c9-43b0-9e10-14cdbb3281ab.png)

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| @react-types/shared   | 3.10.0 |
| @react-aria/button   | 3.3.4 |
| Browser          | Doesn't matter|
| Operating System | Doesn't matter |

**Comment by csantos1113** [comment]
Tracking down the issue:
https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/button/package.json#L25
`@react-aria/button@3.3.4` uses `@react-types/button@3.4.1`
https://github.com/adobe/react-spectrum/blob/main/packages/%40react-types/button/package.json#L12
`@react-types/button@3.4.1` uses `@react-types/shared@3.8.0`

@react-types/shared latest version is 3.10.0: https://github.com/adobe/react-spectrum/blob/main/packages/%40react-types/shared/package.json#L3

`PressEvent` was changed 3 months ago here: https://github.com/adobe/react-spectrum/blame/main/packages/%40react-types/shared/src/events.d.ts#L45 but packages using `@react-types/shared` never got bumped.

**Comment by csantos1113** [comment]
Similar issue here: https://react-spectrum.adobe.com/releases/2021-11-24.html
`@react-aria/toggle` was updated here https://github.com/adobe/react-spectrum/pull/2377/files#diff-b9cf7e178e4b3ff21111866e7a797720843813c9a90165b19a4c28be97f52e9bR71
but `@react-aria/checkbox` and `@react-aria/switch` didn't get published.

so if I use `useCheckbox()` or `useSwitch()` today I get `required` prop instead of the `aria-required` :(

Here is a codesanbox for it: https://codesandbox.io/s/spectrum-checkbox-9wyk2

**Comment by csantos1113** [comment]
## CodeSandbox for `useCheckbox` only

```
"@react-aria/checkbox": "3.2.3",
"@react-stately/toggle": "3.2.3",
```

https://codesandbox.io/s/use-checkbox-rhdqi?file=/src/App.tsx

```
import { useRef } from "react";
import { useCheckbox } from "@react-aria/checkbox";
import { useToggleState } from "@react-stately/toggle";

export default function App() {
  const ref = useRef(null);
  const { inputProps } = useCheckbox(
    { isRequired: true },
    useToggleState({
      defaultSelected: false,
      isReadOnly: false,
      isSelected: false
    }),
    ref
  );
  return <div>{JSON.stringify(inputProps, null, 2)}</div>;
}
```

**Comment by efflam** [comment]
I have a bug with useOption `manager.canSelectItem is not a function` when using `useOption` that seems linked to this issue.
Here a codesandbox to reproduce the issue : It's base on the Select exemple from the docs but updated to use specific packages (@react-aria/..., @react-stately/..., etc)  instead of the globals one.

**Comment by devongovett** [comment]
`@react-aria/checkbox` has a caret range dependency on `@react-aria/toggle`:

https://github.com/adobe/react-spectrum/blob/0571f1cee4565674086ea834327d0d59c093ea1f/packages/%40react-aria/checkbox/package.json#L22

This means it should pick up `@react-aria/toggle` 3.1.5. You may have a lockfile that is preventing this though. Not sure why code sandbox is not picking it up. Maybe they have a cache.

**Comment by csantos1113** [comment]
Hey @devongovett - thanks for reply. I understand what you say about the caret range dependency on the internal packages,
but I still think react-spectrum should do something else to handle this from the code, instead of relaying on consumers to update lockfiles manually?
the codesandox links are the perfect example of the issue: https://codesandbox.io/s/spectrum-checkbox-9wyk2 and https://codesandbox.io/s/spectrum-checkbox-9wyk2  :(

**Comment by efflam** [comment]
I confirm that removing `yarn.lock` and reinstalling dependencies fixed the issue.

**Comment by csantos1113** [comment]
Additionally, for example, during the September release: `@react-aria/button` dependencies were updated ([see file](https://github.com/adobe/react-spectrum/commit/ccb1dec#diff-ec1c5e0817a330c4c70a503bdd6f067c854b35b02e80b9488b85cd32e774a330R21))
![image](https://user-images.githubusercontent.com/9648559/143927453-78eb4cd7-3d1b-4d20-93e1-8325ef613eee.png) even though they were still within the same caret-range

**Comment by devongovett** [comment]
Yeah sometimes I manually bump every package because it's easier than figuring out the minimal set of changes.

I think the current behavior is "correct", but I can understand that it is confusing to deal with lock files. Perhaps we can improve something here.

**Comment by csantos1113** [comment]
For context, these are the steps I had to run to have my dependencies working properly:
```
yarn remove @react-aria/checkbox @react-aria/switch
yarn add @react-aria/checkbox @react-aria/switch
```
Given that checkbox and switch didn't get a new bump, yarn thinks there is nothing to update there (comparing yarn.lock and cache). So I had to delete all dependencies, and re-add them back. this makes `yarn` to pull down latest versions from checkbox and switch internal dependencies.
